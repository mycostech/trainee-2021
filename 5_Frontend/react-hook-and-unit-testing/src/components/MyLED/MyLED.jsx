import React from 'react'

const MyLED = ({ isDanger = false, text }) => {
    return (
        <div
            style={{
                width: '100%',
                height: '60px',
            }}
        >
            <p>
                {isDanger ? 'Danger!' : 'Success'} : {text}
            </p>
        </div>
    )
}

export default MyLED
