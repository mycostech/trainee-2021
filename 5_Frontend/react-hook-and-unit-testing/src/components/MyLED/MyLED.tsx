
interface MyLEDProps {
    isDanger?: boolean
    text?: string
}

function MyLED({
    isDanger,
    text
}: MyLEDProps) {

    return (
        <div
            style={{
                width: '100%',
                height: '60px',
            }}
        >
            <p>
                {isDanger ? 'Danger!' : 'Success'} : {text}
            </p>
        </div>
    )
}

export default MyLED