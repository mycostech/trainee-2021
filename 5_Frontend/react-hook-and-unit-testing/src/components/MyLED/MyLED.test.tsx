import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import MyLED from './MyLED'

describe('MyLED', () => {
    it('render correctly', () => {
        const testText = 'My Computer'
        const tree = renderer.create(<MyLED text={testText} />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('when danger, render correctly with word Danger!', () => {
        const testText = 'My Computer'
        const { getByText, queryByText } = render(
            <MyLED text={testText} isDanger={true} />
        )
        expect(getByText('Danger!', { exact: false })).toBeTruthy()
        expect(queryByText('Success!', { exact: false })).toBeNull()
        expect(queryByText(testText, { exact: false })).toBeTruthy()
    })
})
