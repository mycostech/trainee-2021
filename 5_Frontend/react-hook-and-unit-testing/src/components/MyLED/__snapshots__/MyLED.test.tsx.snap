// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyLED render correctly 1`] = `
<div
  style={
    Object {
      "height": "60px",
      "width": "100%",
    }
  }
>
  <p>
    Success
     : 
    My Computer
  </p>
</div>
`;
